### PR TITLE
Implemented and tested the AsyncMethodsCheck

### PR DESCRIPTION
**Pull request by VarS311**

### CODE REVIEW

File: apex-checks/src/main/java/org/fundacionjala/enforce/sonarqube/apex/checks/AsyncMethodsCheck.java [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+     * It is the code of the rule for the plugin.
+     */
+    public static final String CHECK_KEY = "A1009";
+
+    /**
+     * Value of the annotation future, which this check looks for.
+     */
+    private final String FUTURE = "future";
+
+    private List<AstNode> methods;
```
Comment (JimmyRojas) [inline apex-checks/src/main/java/org/fundacionjala/enforce/sonarqube/apex/checks/AsyncMethodsCheck.java:68]: missing doc

File: apex-checks/src/main/java/org/fundacionjala/enforce/sonarqube/apex/checks/AsyncMethodsCheck.java [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+                            }
+                        }
+                    }
+                }
+            }
+        }
+        return false;
+    }
+
+    private void loadMethods(AstNode ancestor) {
```
Comment (JimmyRojas) [inline apex-checks/src/main/java/org/fundacionjala/enforce/sonarqube/apex/checks/AsyncMethodsCheck.java:146]: missing doc

File: apex-checks/src/main/java/org/fundacionjala/enforce/sonarqube/apex/checks/AsyncMethodsCheck.java (hunk 1 above)
Comment (JimmyRojas) [inline apex-checks/src/main/java/org/fundacionjala/enforce/sonarqube/apex/checks/AsyncMethodsCheck.java:68]: What about use a local variable inside the method ?

**Comment by JimmyRojas** [comment]
see some comments above

**Comment by VarS311** [comment]
@JimmyRojas Fixed the observations, ready to merge

**Comment by JimmyRojas** [comment]
Looks good to me, thank you

**Comment by JimmyRojas** [comment]
@ziur if you don't have some observation, we can merge the changes

**Pull request merged**